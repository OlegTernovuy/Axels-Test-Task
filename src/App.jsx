import "./App.css";
import { Footer, Header } from "./components";
import ProductsPage from "./pages/productsPage/ProductsPage";

function App() {
  return (
    <div className="App">
      <Header/>
      <ProductsPage />
      <Footer/>
    </div>
  );
}

export default App;
