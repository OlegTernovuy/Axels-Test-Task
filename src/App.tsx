import { createBrowserRouter, RouterProvider } from 'react-router-dom';

import ProductsPage from './pages/ProductsPage/ProductsPage';
import NotFound from './pages/NotFound/NotFound';
import { Footer, Header, ProductModal } from './components';
import { AppDiv } from './styled/StyledApp';

const routes = createBrowserRouter([
  {
    path: '/',
    element: <ProductsPage />,
    errorElement: <NotFound />,
    children: [
      {
        path: 'products/:id',
        element: <ProductModal />,
      },
    ],
  },
]);

function App() {
  return (
    <AppDiv>
      <Header />
      <RouterProvider router={routes} />
      <Footer />
    </AppDiv>
  );
}

export default App;
