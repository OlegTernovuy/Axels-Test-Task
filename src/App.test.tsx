import renderer from 'react-test-renderer';

import App from './App';

describe('App snapshotcomponent', () => {
    it('Matches Snapshot', () => {
        const ProductsPageComponent = renderer.create(<App />).toJSON();
        expect(ProductsPageComponent).toMatchSnapshot();
    });
});
