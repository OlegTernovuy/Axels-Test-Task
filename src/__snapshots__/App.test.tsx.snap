// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshotcomponent Matches Snapshot 1`] = `
<div
  className="sc-jlZhRR dwJfrq"
>
  <header
    className="sc-eqUzNf llCMqC"
  >
    <h1>
      Axels Test Task
    </h1>
    <span>
      Oleh Ternovyi
    </span>
  </header>
  <div>
    This page does not exist
  </div>
  <footer
    className="sc-fqkwJk eedRsB"
  >
    <span>
      Oleh Ternovyi 2024
    </span>
  </footer>
</div>
`;
